packages/text-input/src/TextInput.tsx

```diff
@@ -371,6 +382,7 @@ const TextInput: React.ComponentType<React.PropsWithRef<
               onChange={onValueChange}
               ref={forwardRef}
               id={id}
+              autoComplete={disabled ? 'off' : 'on'}
```

Comment:
This doesn't support the hinting we should do in forms when it's on.

The logic should probably look like:

```
autoComplete={disabled ? 'off' : props.autoComplete || null}
```